### PR TITLE
feat: return prev and next post IRI

### PR DESCRIPTION
**Pull request by blushi**
## Description

Part of: https://regennetwork.atlassian.net/browse/APP-194?atlOrigin=eyJpIjoiNzM5NzkzNmI1M2VkNDI2Zjg0ZjFmZDA1YjE1MWIxNmUiLCJwIjoiaiJ9

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] targeted `dev` branch
- [x] provided a link to the relevant issue or specification
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed
- [ ] submit a PR against `master` branch after this one (and other PRs depending on this one, e.g. on regen-network/regen-web, if applicable) get merged

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**.*

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

**Review by paul121 — APPROVED** [review]
LGTM but one question: What should the behavior be if the previous or next post are `private` and the current user doesn't have access?

I'm not sure what the design spec is, if the app will still display that these private posts exist but not show any of the data then this is probably fine. Otherwise it might be nice if the previous and next buttons only appear for posts that the user can access.

**Comment by blushi** [comment]
> LGTM but one question: What should the behavior be if the previous or next post are private and the current user doesn't have access?

The user would see the following page after clicking prev or next: https://www.figma.com/design/Bksz1JeDYxQVIXdI46EgPT/Project-Posts?node-id=1053-169368&m=dev

There's no prev and next button on this page, meaning the user won't be able to navigate further with prev/next from there. Maybe we want to add such buttons, @erikalogie?

**Comment by erikalogie** [comment]
> > LGTM but one question: What should the behavior be if the previous or next post are private and the current user doesn't have access?
> 
> The user would see the following page after clicking prev or next: https://www.figma.com/design/Bksz1JeDYxQVIXdI46EgPT/Project-Posts?node-id=1053-169368&m=dev
> 
> There's no prev and next button on this page, meaning the user won't be able to navigate further with prev/next from there. Maybe we want to add such buttons, @erikalogie?

We could, or we could just skip any pages that are private like this. Wdyt @clevinson?

**Comment by blushi** [comment]
> > > LGTM but one question: What should the behavior be if the previous or next post are private and the current user doesn't have access?
> > 
> > 
> > The user would see the following page after clicking prev or next: https://www.figma.com/design/Bksz1JeDYxQVIXdI46EgPT/Project-Posts?node-id=1053-169368&m=dev
> > There's no prev and next button on this page, meaning the user won't be able to navigate further with prev/next from there. Maybe we want to add such buttons, @erikalogie?
> 
> We could, or we could just skip any pages that are private like this. Wdyt @clevinson?

After some discussion with @clevinson we decided to skip the any private posts (if not project admin).
Ready for a final look.

File: server/__tests__/e2e/posts.get.test.ts [hunk 1]
```diff
@@ -5,27 +5,62 @@ import {
...
-    expect(data.iri).toEqual(iri);
+    expect(authData.iri).toEqual(iri);
+    expect(authData.prevIri).toEqual(prevIri);
+    expect(authData.nextIri).toEqual(nextIri);
+
+    // Get post with any app user
+    const resp = await fetch(`${getMarketplaceURL()}/posts/${iri}`, {
+      method: 'GET',
+    });
+    const data = await resp.json();
```
Comment (paul121) [inline server/__tests__/e2e/posts.get.test.ts:60]: Maybe add a check to make sure this was a successful request?

```suggestion
    expect(resp.status).toBe(200);
    const data = await resp.json();
    expect(data.iri).toEqual(iri);
```

**Review by paul121 — APPROVED** [review]
Just a small suggestion but looks good

**Pull request merged**